src/explorer/CosmosDBDatabase.ts

```diff
@@ -22,48 +33,52 @@ export class CosmosDBDatabase extends AzureTreeItem<ISiteTreeRoot> {
         this.label = this.getLabel(connectionId);
     }
 
-    public get iconPath(): string | vscode.Uri | { light: string | vscode.Uri; dark: string | vscode.Uri } {
-        return {
-            light: path.join(__filename, '..', '..', '..', '..', 'resources', 'light', 'Database.svg'),
-            dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'dark', 'Database.svg')
-        };
-    }
-
     public async deleteTreeItemImpl(): Promise<void> {
-        const connectionToDelete = this.connectionId;
-        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
-        const connections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
-        const connectionsUnit = connections.find((x: IConnections) => x.webAppId === this.root.client.id);
-        if (connectionsUnit && connectionsUnit.cosmosDB) {
-            const indexToDelete = connectionsUnit.cosmosDB.findIndex((x: string) => x === connectionToDelete);
-            if (indexToDelete > -1) {
-                connectionsUnit.cosmosDB.splice(indexToDelete, 1);
-                workspaceConfig.update(constants.configurationSettings.connections, connections);
-            }
-        }
+        throw new Error('Not implemented yet!');
     }
 
     private getLabel(id: string): string {
-        const items = this.parseCosmos(id) || this.parseAttached(id);
-        if (!items) {
-            throw new Error('Failed to parse connection id');
+        const items = this.mongoGetAccountDatabase(id);
```

Comment:
The Cosmos DB extension already has logic to create a label from the connection string. See `getServerIdFromConnectionString` here:
https://github.com/Microsoft/vscode-cosmosdb/blob/master/src/tree/AttachedAccountsTreeItem.ts#L211

It's completely different than what you're doing and I don't want to duplicate that logic, so this should probably be included in the API. Maybe you pass in the connection string and it just returns the account name and database name

Reply:
I think it's better to use something like parent.label(in Cosmos API method). As I understand, for attached this will give us the same label as `getServerIdFromConnectionString `(the label is already created by that function), but for "internal" we'll get account labels that are actually in CosmosDB explorer. 
Oh, will I have those labels for attached databases if I didn't "loadmore"  "Attached Database Accounts" node? 

Wanted to parse connection string because wasn't sure that it's a good idea to make API calls for all connections when loadmore method is running. So, if we do not care a lot about API calls can we just get Cosmos treeItemId from getDatabase() method(when the loadmore method is running)? In this case, Reveal command will stay the same and we don't need to overcomplicate it.